Main.lean

```diff
@@ -376,8 +388,8 @@ def main : IO Unit := do
       (param i32)
       (result i32 i32) (result i32 i32)
 
-      (i32.const 1)
-      (nop)
+      (block (result i32) (i32.const 0))
+      (if (result i32) then (i32.const 8) else (i32.const 2))
```

Comment:
What does result mean here?

Reply:
The result type of the branches. It should actually support type indices, but for now it's just inlined value types. See [spec](https://webassembly.github.io/spec/core/syntax/instructions.html#syntax-blocktype)

Reply:
TIL. Great job!